tests/test_stream.py

```diff
@@ -24,6 +44,45 @@ def test_pretty_redirected_stream(httpbin):
     assert BINARY_SUPPRESSED_NOTICE.decode() in r
 
 
+def test_pretty_stream_ensure_full_stream_is_retrieved(httpbin):
+    env = MockEnvironment(
+        stdin=StdinBytesIO(),
+        stdin_isatty=False,
+        stdout_isatty=False,
+    )
+    r = http('--pretty=format', '--stream', 'GET',
+             httpbin.url + '/stream/3', env=env)
+    assert r.count('/stream/3') == 3
+
+
+@pytest.mark.parametrize('pretty', PRETTY_OPTIONS)
+@pytest.mark.parametrize('stream', [True, False])
+@responses.activate
+def test_pretty_options_with_and_without_stream_with_converter(pretty, stream):
+    plugin_manager.register(FakeConverterPlugin)
+    try:
+        assert 'FakeConverterPlugin' in str(plugin_manager)
+
+        url = 'http://example.org'  # Note: URL never fetched
+        body = b'\x00{"foo":42,\n"bar":"baz"}'
+        responses.add(responses.GET, url, body=body,
+                      stream=True, content_type='json/bytes')
+
+        args = ['--pretty=' + pretty, 'GET', url]
+        if stream:
+            args.insert(0, '--stream')
+        r = http(*args)
+
+        assert 'json/bytes' in r
+        if pretty == 'none':
+            assert BINARY_SUPPRESSED_NOTICE.decode() in r
+        else:
+            assert '"bar": "baz",' in r
+            assert '"foo": 42' in r
```

Comment:
The JSON object [should be accessible](https://github.com/httpie/httpie/blob/350f973f705dc996af27d66fe02e03e2f1f99968/tests/utils/__init__.py#L139-L171) via `r.json`. Is it not working in this case? If it is, then this is preferable:

```python
assert r.json['bar'] == 'baz'
```

Reply:
I see! This comment would make a useful `# comment`.